src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3282,6 +3289,24 @@ TransactionStateMachine(WorkerSession *session)
 					break;
 				}
 
+				/* if this is a multi-query task, send the next query */
+				if (placementExecution->queryIndex < task->queryCount)
```

Comment:
Do we ever need to clear results first? or is sending multiple query strings the only way we can multiple results?

Reply:
I couldn't find this it by looking at libpq's or postgres protocol's documentation. Clearing results wouldn't hurt, so added it.

Reply:
I re-read [postgresql's protocol's desceiption](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-FLOW-EXT-QUERY). For parameterized queries it uses the extended query protocol, in which successful execution is terminated by appearance of exactly one `CommandComplete`, which translates to `PGRES_COMMAND_OK` in libpq, and then it switches state to wait for the next query.

So we don't need need to clear results here.

Reply:
Then we shouldn't clear results, since that blocks the executor.